### PR TITLE
Audit API -Peer Review - Direct commits with Increment version tag by Service Account 

### PR DESCRIPTION
**Pull request by nireeshT**
Audit API -Peer Review - Direct commits with Increment version tag by service account /non- service account

### CODE REVIEW

File: api-audit/src/main/java/com/capitalone/dashboard/ApiSettings.java [hunk 1]
```diff
@@ -18,6 +18,7 @@
     private String peerReviewContexts;
     private String peerReviewApprovalText;
     private String serviceAccountOU;
+    private String incrementVersionTagRegEx;
```
Comment (tabladrum) [inline api-audit/src/main/java/com/capitalone/dashboard/ApiSettings.java]: may be keep it generic? "commitLogIgnoreAuditRegEx"?

File: api-audit/src/main/java/com/capitalone/dashboard/evaluator/CodeReviewEvaluator.java [hunk 2]
```diff
@@ -186,12 +189,38 @@ private CodeReviewAuditResponseV2 getPeerReviewResponses(CollectorItem repoItem,
             if (!allPrCommitShas.contains(commit.getScmRevisionNumber()) &&
                     StringUtils.isEmpty(commit.getPullNumber()) && commit.getType() == CommitType.New) {
                 commitsNotDirectlyTiedToPr.add(commit);
-                if (!StringUtils.isEmpty(commit.getScmAuthorLDAPDN()) && CommonCodeReview.checkForServiceAccount(commit.getScmAuthorLDAPDN(), settings))  {
-                    reviewAuditResponseV2.addAuditStatus(CodeReviewAuditStatus.COMMITAUTHOR_EQ_SERVICEACCOUNT);
-                }
+                // auditServiceAccountChecks includes - check for service account and increment version tag for service account on direct commits.
+                auditServiceAccountChecks(reviewAuditResponseV2, commit);
+                // check for increment version tag and flag Direct commit by Non Service account
+                auditNonServiceAccountChecks(reviewAuditResponseV2, commit);
```
Comment (tabladrum) [inline api-audit/src/main/java/com/capitalone/dashboard/evaluator/CodeReviewEvaluator.java]: combine these two methods?

File: api-audit/src/main/java/com/capitalone/dashboard/evaluator/CodeReviewEvaluatorLegacy.java [hunk 3]
```diff
@@ -194,5 +195,29 @@ private CodeReviewAuditResponse getErrorResponse(CollectorItem repoItem, String
...
+    }
+
+    private void auditNonServiceAccountChecks(CodeReviewAuditResponse codeReviewAuditResponse, Commit commit) {
+        if (!isServiceAccount(commit.getScmAuthorLDAPDN()))  {
+           auditIncrementVersionTag(codeReviewAuditResponse, commit, CodeReviewAuditStatus.DIRECT_COMMIT_INCREMENT_VERSION_TAG_NON_SERVICE_ACCOUNT);
+        }
+    }
+
+    private boolean isServiceAccount(String scmAuthorLDAPDN) {
+        return !StringUtils.isEmpty(scmAuthorLDAPDN) && CommonCodeReview.checkForServiceAccount(scmAuthorLDAPDN, settings);
```
Comment (tabladrum) [inline api-audit/src/main/java/com/capitalone/dashboard/evaluator/CodeReviewEvaluatorLegacy.java]: I think the checkForServiceAccount already checks for empty string. In which case, we don't need this method?

File: api-audit/src/main/java/com/capitalone/dashboard/ApiSettings.java (hunk 1 above)
Comment (nireeshT) [reply api-audit/src/main/java/com/capitalone/dashboard/ApiSettings.java]: @tabladrum  done

File: api-audit/src/main/java/com/capitalone/dashboard/evaluator/CodeReviewEvaluator.java (hunk 2 above)
Comment (nireeshT) [reply api-audit/src/main/java/com/capitalone/dashboard/evaluator/CodeReviewEvaluator.java]: @tabladrum  done

File: api-audit/src/main/java/com/capitalone/dashboard/evaluator/CodeReviewEvaluatorLegacy.java (hunk 3 above)
Comment (nireeshT) [reply api-audit/src/main/java/com/capitalone/dashboard/evaluator/CodeReviewEvaluatorLegacy.java]: @tabladrum  - done

**Pull request merged**